src/main/java/org/walkmod/javalang/compiler/types/TypesLoaderVisitor.java

```diff
@@ -165,12 +165,23 @@ static String resolveSymbolName(final String name, final boolean imported, final
     }
 
     public void setClassLoader(ClassLoader cl) {
-        if (classLoader.getParent() != cl) {
-            classLoader = new SymbolTypesClassLoader(cl);
+        if (applicationClassLoader != cl) {
+            applicationClassLoader = cl;
+            if (cl instanceof URLClassLoader) {
+                URLClassLoader aux = (URLClassLoader) cl;
+                IndexedURLClassLoader icl = new IndexedURLClassLoader(aux.getURLs(), aux.getParent());
+                classLoader = new CachedClassLoader(icl);
+
+            } else {
+                classLoader = new CachedClassLoader(new IndexedURLClassLoader(cl));
+            }
+            SDKFiles = classLoader.getSDKContents("java.lang");
```

Comment:
Do you care for avoiding statics at the moment or do you want deal with them later?
There is some kind of context missing that brackets an execution/run.
I am dreaming of parallel execution here...

Reply:
Really good point about the parallel execution here :)

Well, I think that I cannot add more statics for now without overcomplicating the code. Indeed with visualVM now, the hotspot is the `LoadStaticImportsAction` because of the reflection usage there asking for the import class. 

What do you mean with a missing context?

Reply:
If you want to do parallel runs each run needs it's own set of mutable state.
Often ThreadLocals are used but I prefer explicit passing of a kind of "execution context" that stores data global for a run. For single threaded apps the "statics" provide that.

Reply:
Cal, let's work in the parallel part in a different PR. I do not see how to do it easily with the current logic.

Reply:
Sure. I don't want to *work* on it now. I just wanted to point out not working too much *against* it.